src/Microsoft.PowerShell.Commands.Management/commands/management/NewPropertyCommand.cs

```diff
@@ -175,4 +182,122 @@ protected override void ProcessRecord()
         #endregion Command code
 
     }
+
+    /// <summary>
+    /// Provides argument completion for PropertyType parameter.
+    /// </summary>
+    public class PropertyTypeArgumentCompleter : IArgumentCompleter
+    {
+        private static readonly OrderedDictionary s_RegistryCompletionResults = new()
+        {
+            { "String", "A normal string." },
+            { "ExpandString", "A string that contains unexpanded references to environment variables that are expanded when the value is retrieved." },
+            { "Binary", "Binary data in any form." },
+            { "DWord", "A 32-bit binary number." },
+            { "MultiString", "An array of strings." },
+            { "QWord", "A 64-bit binary number." },
+            { "Unknown", "An unsupported registry data type." }
+        };
+
+        /// <summary>
+        /// Returns completion results for PropertyType parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            // -PropertyType parameter is only supported on Windows
+            if (!Platform.IsWindows)
+            {
+                yield break;
+            }
+
+            Collection<PathInfo> paths;
+
+            if (fakeBoundParameters.Contains("Path"))
+            {
+                paths = ResolvePaths(
+                    ConvertParameterPathsToArray(fakeBoundParameters["Path"]),
```

Comment:
Really we have no need to get all paths.
We need only `paths[0].Provider.NameEquals("Registry")`.
I'd expect `IsRegistryProvider(fakeBoundParameters)`.

Reply:
Thanks. I have removed the array converting in 17ec6dc0e.